src/details/ArborX_DetailsTreeTraversal.hpp

```diff
@@ -97,41 +100,58 @@ struct TreeTraversal<BVH, Predicates, Callback, SpatialPredicateTag>
 
     Node const *stack[64];
     Node const **stack_ptr = stack;
-    *stack_ptr++ = nullptr;
-    Node const *node = _bvh.getRoot();
+    *stack_ptr++ = nullptr;            // push sentinel value
```

Comment:
Would prefer to use something than `sentinel` here, as we already use it in a different context for ropes. Something like `termination value` should work.